gobblin-yarn/src/main/java/org/apache/gobblin/yarn/YarnAutoScalingManager.java

```diff
@@ -260,12 +302,14 @@ void runInternal() {
 
       // adjust the number of target containers based on the configured min and max container values.
       numTargetContainers = Math.max(this.minContainers, Math.min(this.maxContainers, numTargetContainers));
+      trimContainerSize(numTargetContainers, yarnContainerRequestBundle);
```

Comment:
I am not sure we should dynamically trim container size. Will lead to unpredictable behavior at runtime. I think we should fail fast if Yarn cannot meet our container request requirements. This is the behavior in YarnService.

Reply:
This is only to ensure the container size is within the range of max and min container size read from config. Originally the resource requirement was universal for all containers. Since we have different resource requirement for different helix tag, I'm not sure there's any better way.

Reply:
Shouldn't we have this issue even without the changes in this PR? How is it handled currently? IIRC, the resource request just fails if it exceeds the max specified by Yarn.